src/neo-vm/ExecutionEngine.cs

```diff
@@ -370,8 +370,14 @@ private void ExecuteInstruction()
                 case OpCode.ASSERT:
                     {
                         var x = Pop().GetBoolean();
+                        string? msg = "";
+                        if (String.CompareOrdinal(CurrentContext.Compiler, "nccs v3.0.3") > 0)
```

Comment:
Compiler-specific behavior in VM?

Reply:
> Compiler-specific behavior in VM?

Yep, cause I added one extra parameter to `assert`, this is incompatible with contracts compiled by the previous compiler.